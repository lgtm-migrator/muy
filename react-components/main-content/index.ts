export { MainContent } from './main-content';
export type { MainContentProps } from './main-content';
