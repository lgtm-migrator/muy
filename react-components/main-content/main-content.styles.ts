import { createStyles, makeStyles } from "@material-ui/core"

const styles = createStyles({
  root: {
    display: "block",
    outlineColor: "transparent",
    outlineStyle: "none",
    outlineWidth: 0,
  }
});

export const useStyles = makeStyles(styles, { name: "MainContent" });