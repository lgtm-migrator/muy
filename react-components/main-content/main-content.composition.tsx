import * as React from 'react';
import { MainContent } from './main-content';

export const NoPropsMainContent = () => (
  <MainContent />
);

export const BasicMainContent = () => (
  <MainContent>Hello from MainContent</MainContent>
);
