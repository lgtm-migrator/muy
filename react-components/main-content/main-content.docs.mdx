---
labels: ['react', 'typescript', 'main-content']
description: 'A MainContent component.'
---

import { MainContent } from './main-content';

MainContent example:

A simple MainContent component with some text

Code Snippet:
```js
<MainContent id={"code-snippet"}>
  Hello from MainContent
</MainContent>
```


Live Playground:
```js live
<MainContent id={"live-playground"}>
  Hello from MainContent
</MainContent>
```
