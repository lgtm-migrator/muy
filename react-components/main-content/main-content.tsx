import * as React from "react";
import * as PropTypes from "prop-types";
import classNames from "classnames";
import { useStyles } from "./main-content.styles"

export type MainContentProps = {
  children?: React.ReactText,
  className?: string;
  id?: string;
};

export const MainContent = (props: MainContentProps) => {

  const {
    children,
    className,
    id = "content",
  } = props;

  const { root } = useStyles();

  return (
    <main
      className={classNames(root, className)}
      id={id}
      role={"main"}
      tabIndex={-1}>
      {children}
    </main>
  );
}

MainContent.defaultProps = {
  id: "content"
};

MainContent.displayName = "MainContent";

MainContent.propTypes = {
  id: PropTypes.string
}