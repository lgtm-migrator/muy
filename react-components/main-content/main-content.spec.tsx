import * as React from 'react';
import { create } from "react-test-renderer";
import { render } from '@testing-library/react';
import { BasicMainContent, NoPropsMainContent } from "./main-content.composition"

describe('main-content', () => {
  describe("with no props", () => {
    it("should match snapshot", () => {
      const rendered = create(<NoPropsMainContent />);
      expect(rendered).toMatchSnapshot();
    })
  })
  it('should render with the correct text', () => {
    const { getByText } = render(<BasicMainContent />);
    const rendered = getByText('Hello from MainContent');
    expect(rendered).toBeTruthy();
  });
});
