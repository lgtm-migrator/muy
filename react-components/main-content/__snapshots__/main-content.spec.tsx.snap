// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main-content with no props should match snapshot 1`] = `
<main
  className="MainContent-root-1"
  id="content"
  role="main"
  tabIndex={-1}
/>
`;
